pyqt5-gui/main_menu.py

```diff
@@ -120,8 +130,8 @@ def books_open():
                     category_description.setFont(QFont(fonts[0], 9))
                     category_button = QPushButton()
                     current_function_name = categories_functions.popleft()
-                    current_function = getattr(food_open(), current_function_name)
-                    category_button.clicked.connect(lambda: current_function())
+                    print(current_function_name)
+                    category_button.clicked.connect(lambda: functions_dict[current_function_name]())
                     current_vertical_layout.addWidget(category_name)
                     current_vertical_layout.addWidget(category_description)
                     current_vertical_layout.addWidget(category_button)
```

Comment:
The idea of making the categories groupboxes looks very good. We can add a font in the directory "fonts" and apply it to all the text in the categories_groupbox. Can you show me how can I add another fonts and where can I find them? I will apply it to all the text in categories_groupbox, if you are okay with it.

Reply:
For new fonts you can use [dafont.com](https://www.dafont.com/), but there are hundreds of web sites where you can download fonts for free, [Google Fonts](https://fonts.google.com/) is also a good place to take a look.

After you download the font, you need to place it in `fonts` directory, the file is with extention `.otf`, `.ttf`...depending on the font license. Then you can import the font inside the application using the Qt Framework.


```python
"""This is the old JetBrains Font we are using for most of the objects"""
font = QFontDatabase.addApplicationFont(r'/fonts/jetbrains-mono.regular.ttf') 
"""This would be the new font"""
font = QFontDatabase.addApplicationFont(r'/path/to/the/new_font') 
if font >= 0: """<-- Change this to only greather than. If we have 0 custom fonts, it should throw an error"""
   fonts = QFontDatabase.applicationFontFamilies(font)
else:
   print("Error loading fonts!")
```

When setting a custom font to any QLabel() object, you can call the `fonts` array and specify which font exactly you want to use:

```python
some_label = QLabel()
some_label.setText("This is some text label")
some_label.setFont(QFont(fonts[n], 23)) """Where n is the exact fonts array index, in our case 0 or 1"""
```

This should work 🤔